pkg/admissioncontroller/webhook/admission/resourcesize/handler.go

```diff
@@ -105,6 +132,43 @@ func (h *Handler) handle(req admission.Request) error {
 	return nil
 }
 
+func (h *Handler) handleCountLimit(req admission.Request, log logr.Logger, requestedResource *metav1.GroupVersionResource, count *int64) error {
+	if req.Namespace != "" {
+		// We only want to restrict non-namespaced resources
+		// namespaced resources can be restricted by ResourceQuotas
+		log.Info("Skipping count check for namespaced resource", "resource", requestedResource.String())
+		return nil
+	}
+
+	// Only count for CREATE operations, not UPDATE
+	if req.Operation != admissionv1.Create {
+		return nil
+	}
+
+	exceedsLimit, err := h.existingResourcesExceedLimit(req, requestedResource, ptr.Deref(count, 0))
+	if err != nil {
+		log.Error(err, "Failed to determine if count exceeds limit", "resource", requestedResource.String())
+		return err
```

Comment:
```suggestion
		return fmt.Errorf("failed to determine if count exceeds limit: %w", err)
```

I assume proper logging is already done by the Controller-Runtime in case of an error?!